flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/GraphMetrics.java

```diff
@@ -150,20 +149,15 @@ private static boolean parseParameters(String[] args) {
 	}
 
 	@SuppressWarnings("serial")
-	private static DataSet<Edge<Long, NullValue>> getEdgesDataSet(ExecutionEnvironment env) {
-		if (fileOutput) {
-			return env.readCsvFile(edgesInputPath)
-					.lineDelimiter("\n").fieldDelimiter("\t")
-					.types(Long.class, Long.class).map(
-							new MapFunction<Tuple2<Long, Long>, Edge<Long, NullValue>>() {
-
-								public Edge<Long, NullValue> map(Tuple2<Long, Long> value) {
-									return new Edge<Long, NullValue>(value.f0, value.f1, 
-											NullValue.getInstance());
-								}
-					});
-		} else {
-			return ExampleUtils.getRandomEdges(env, NUM_VERTICES);
+	private static Graph<Long, NullValue, NullValue> getGraph(ExecutionEnvironment env) {
+		if(fileOutput) {
+			return Graph.fromCsvReader(edgesInputPath, env).lineDelimiterEdges("\n").fieldDelimiterEdges("\t")
+										.types(Long.class);
+
+		}
+		else
+		{
+			return Graph.fromDataSet(ExampleUtils.getRandomEdges(env, NUM_VERTICES), env);
```

Comment:
Yup... so I like how this looks better than how the previous rewritings were made...